File: src/test/java/com/planbase/pdf/layoutmanager/SingleItemRenderatorTest.java
package com.planbase.pdf.layoutmanager;

import org.junit.Test;
import org.organicdesign.fp.oneOf.Option;

import static org.junit.Assert.*;

public class SingleItemRenderatorTest {
    @Test public void testSingleItemRenderator() {
        Renderable r = new Renderable() {
            @Override public XyDim calcDimensions(float maxWidth) { return XyDim.of(5f, 7f); }

            @Override public XyOffset render(RenderTarget lp, XyOffset outerTopLeft, XyDim outerDimensions) {
                return null;
            }

            @Override public Renderator renderator() {
                return new Renderator.SingleItemRenderator(this);
            }
        };
        Renderator tor = r.renderator();
        assertTrue(tor.hasMore());
        assertEquals(r.calcDimensions(0), tor.getSomething(9f).xyDim());
        assertFalse(tor.hasMore());

        tor = r.renderator();
        assertTrue(tor.hasMore());
        assertEquals(Option.NONE, tor.getIfFits(3f));
        assertTrue(tor.hasMore());
        assertEquals(r.calcDimensions(0), tor.getIfFits(9f).get().xyDim());
        assertFalse(tor.hasMore());
    }
}

File: src/test/java/com/planbase/pdf/layoutmanager/SingleItemRenderatorTest.kt
package com.planbase.pdf.layoutmanager

import org.junit.Test

import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class SingleItemRenderatorTest {
    @Test fun testSingleItemRenderator() {
        val r = object : Renderable {
            override fun calcDimensions(maxWidth: Float): XyDim {
                return XyDim.of(5f, 7f)
            }

            override fun render(lp: RenderTarget, outerTopLeft: XyOffset, outerDimensions: XyDim): XyOffset? {
                return null
            }

            override fun renderator(): Renderator {
                return Renderator.SingleItemRenderator(this)
            }
        }

        var tor = r.renderator()
        assertTrue(tor.hasMore())
        assertEquals(r.calcDimensions(0f), tor.getSomething(9f).match({it}, {it}).xyDim())
        assertFalse(tor.hasMore())

        tor = r.renderator()
        assertTrue(tor.hasMore())
        assertTrue(tor.getIfFits(3f).isNone)
        assertTrue(tor.hasMore())
        assertEquals(r.calcDimensions(0f), tor.getIfFits(9f).match({ c -> c }, { t -> t }, null).xyDim())
        assertFalse(tor.hasMore())
    }
}